### PR TITLE
Implemented mutation to change column to category mapping

### LINKED ISSUE #252: create mutation to change column to category mapping
When called with arguments for a `column` and a `category`, this mutation should change the link between a `column` and a `category` in the following ways:

- [x] if the `column` and `category` are already "linked", the `column` should be "unlinked" from this `category`, i.e. set to a default value indicating "no link"
- [x] else: if the `column` is not yet linked to any category OR is linked to a different category than `category`, the `column` should now be linked to `category`

Other requirements:

- [x] depending components have been checked and are calling this mutation correctly
- [x] depending component tests have been checked and are calling this mutation correctly
- [x] have unit tests to check functionality
- [x] needed store state variables exist already or have been created if not

### PR DESCRIPTION
**Pull request by rmanaem**
closes #252

### CODE REVIEW

File: store/index-refactor.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    // columnToCategoryMapping: {}
+});
+
+export const getters = {
+
+};
+
+
+export const mutations = {
+    alterColumnCategoryMapping(p_state, activeCategory, columnName) {
```
Comment (surchs) [inline store/index-refactor.js:14]: minor point, but I think in the context of the store `activeCategory` isn't that intuitive anymore. I guess `targetCategory` would be clearer?

You could also consider adding a top level docstring to the mutation to summarize the intention of it (link a column to a new category, unless it is already linked - or something like that)

**Review by surchs — CHANGES_REQUESTED** [review]
Very cool PR @rmanaem :tada:, thank you! Thanks also for fixing that little extra column thing in the `column-linking-table`, I think I had missed that. 

Everything looks good, I left some comments to point out things I think you left over from debugging. And I think the store mutation either needs a docstring or a slightly clearer argument name for the category - or maybe both. 

With these changes: good to go!

edit: also: very cool that our cypress runner just picks up the new `unit` tests! Eventually we could look into having those run as part of their own action!

File: cypress/component/column-linking-table.cy.js [hunk 2]
```diff
@@ -87,15 +86,16 @@ describe("Tests basic functionality of the table that links categories with data
         // 0. The first category and column
         const participantIDColumn = store.getters.columns()[0].name;
         const subjectIDCategory = store.getters.categories()[0];
+        console.log(participantIDColumn, subjectIDCategory);
 
         // 1. Arrange - Set up the spy, mount the component, and bind the spy to it
         const onColumnNameClickedSpy = cy.spy().as("onColumnNameClickedSpy");
```
[the remark this answers is not in the document]
Comment (rmanaem) [reply cypress/component/column-linking-table.cy.js:92]: Yea good catch! Forgot to remove the "break point".

File: cypress/component/column-linking-table.cy.js [hunk 3]
```diff
@@ -87,15 +86,16 @@ describe("Tests basic functionality of the table that links categories with data
         // 0. The first category and column
         const participantIDColumn = store.getters.columns()[0].name;
         const subjectIDCategory = store.getters.categories()[0];
+        console.log(participantIDColumn, subjectIDCategory);
```
[the remark this answers is not in the document]
Comment (rmanaem) [reply cypress/component/column-linking-table.cy.js:89]: Yea :sweat_smile:

File: cypress/component/annot-columns.cy.js [hunk 4]
```diff
@@ -16,7 +16,7 @@ const store = {
         }
     },
     mutations: {
-        removeColumn: () => (activeCategory, columnName) => {}
```
[the remark this answers is not in the document]
Comment (rmanaem) [reply cypress/component/annot-columns.cy.js:19]: 🙏🏼

File: cypress/unit/store-mutation-alterColumnCategoryMapping.cy.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            "column2": null,
+            "column3": "Sex",
+            "column4": null,
+            "column5": null
+        }
+    };
+    alterColumnCategoryMapping(state, "someCategory", "column1");
+    expect(state.columnToCategoryMapping.column1).to.equal("someCategory");
+    alterColumnCategoryMapping(state, "someCategory", "column2");
+    expect(state.columnToCategoryMapping.column2).to.equal("someCategory");
```
[the remark this answers is not in the document]
Comment (rmanaem) [reply cypress/unit/store-mutation-alterColumnCategoryMapping.cy.js:32]: :pray:

File: store/index-refactor.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+// Facilitate Vue reactivity via 'Vue.set' and 'Vue.delete'
+import Vue from "vue";
+
+export const state = () => ({
+    // columnToCategoryMapping: {}
```
[the remark this answers is not in the document]
Comment (rmanaem) [reply store/index-refactor.js:5]: Yea thanks for mentioning it, I recall discussing this. Definitely forgot to uncomment.

File: store/index-refactor.js (hunk 1 above)
Comment (rmanaem) [reply store/index-refactor.js:14]: Good call. Will do both.

**Pull request merged**